src/js/uri_scheme.js

```diff
@@ -0,0 +1,75 @@
+import $ from 'jquery';
+
+import GUI from './gui';
+import serial from './serial';
+import PortHandler from './port_handler';
+import CONFIGURATOR from './data_storage.js';
+
+export async function handleURIScheme() {
+    console.log(`Listening for ${CONFIGURATOR.uriScheme}:// URI scheme invocations/events`);
+
+    if (GUI.isNWJS()) {
+        handleNWJSArgv();
+        handleNWJSOpenEvents();
+    } else if (GUI.isCordova()) {
+        handleCordovaOpenEvents();
+    }
+}
+
+function connectTcp(connectionString, eventType) {
+    if (!connectionString) {
+        return;
+    }
+
+    console.log(`Processing ${eventType} with connection string "${connectionString}"`);
+
+    const protocolScheme = `${CONFIGURATOR.uriScheme}://`;
+    // NOTE: On some platforms a trailing slash is added automatically (it needs to be excluded)
+    const connectionStringRegex = new RegExp(`^.*${protocolScheme}(.+?)/?$`);
+
+    const url = connectionString.match(connectionStringRegex)?.[1];
+    if (!url) {
+        console.log(`Invalid ${protocolScheme} connection string "${connectionString}" from ${eventType}`);
+        return;
+    }
+
+    const tcpUrl = `tcp://${url}`;
+    if (!serial.tcpUrlRegex.exec(tcpUrl)) {
+        console.log(`Invalid ${protocolScheme} TCP URL ${tcpUrl} (from ${eventType} with connection string "${connectionString}")`);
+        return;
+    }
+
+    console.log(`Connecting to ${tcpUrl} (from ${eventType} with connection string "${connectionString}")`);
+
+    // Explicitly add the 'manual' select option if the port options haven't loaded yet
+    // (e.g. during a cold start)
+    if ($('#port [value="manual"]').length === 0){
+        PortHandler.addManualPortSelectOption();
+    }
+
+    $('#port').val('manual').trigger('change');
+    $('#port-override').val(tcpUrl).trigger('change');
+    $('div.connect_controls a.connect').trigger('click');
+}
+
+
+function handleNWJSArgv() {
+    const connectionString = nw?.App?.argv?.[0] || '';
+
+    if (connectionString) {
+        connectTcp(connectionString, 'cold start');
+    }
+}
+
+function handleNWJSOpenEvents() {
+    nw?.App?.on('open', (connectionString) => {
+        connectTcp(connectionString, 'NW.js "open" event');
+    });
```

Comment:
When is this fired exactly?

Reply:
@chmelevskij The `open` event is fired when the URI scheme is invoked while the configurator is already running. [As per the NW.js docs](https://docs.nwjs.io/en/latest/References/Command%20Line%20Options/#about-command-line-arguments):

> If there is already one running instance of your app, the full command line of the second instance will be passed to the existing instance by sending open event to the App object.

Coming back [to your earlier comment](https://github.com/betaflight/betaflight-configurator/pull/3831#discussion_r1525475582) - it seems that the event handler is always called with a single string which contains the executable path along with all CLI arguments - e.g. if I run `betaflight-configurator --someTestArg btfltcp://10.0.0.2` while the configurator is running I see a `connectionString` value similar to:

```
/opt/betaflight/betaflight-configurator/betaflight-configurator --someTestArg --disable-features=nw2 --user-data-dir=/home/USER/.config/betaflight-configurator --no-sandbox --no-zygote --flag-switches-begin --flag-switches-end --nwapp=/opt/betaflight/betaflight-configurator btfltcp://10.0.0.2
```
(the behavior is indeed a bit inconsistent in regards to `nw.App.argv` and apparently there is no filtering of the NW.js arguments - so that's why I went with for regex which extract just the `btfltcp://{ip}` segment)

Reply:
Ah, fair, ignore my previous comment. If we have something to just check that url is correct seems fine to me 😊